File: gitdot-web/packages/api/src/endpoint/review/get-review-diff.ts
import { z } from "zod";
import { RepositoryDiffResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const GetReviewDiffResponse = z.object({
  files: z.array(RepositoryDiffResource),
});
export type GetReviewDiffResponse = z.infer<typeof GetReviewDiffResponse>;

export const GetReviewDiff = {
  path: "/repository/{owner}/{repo}/review/{number}/diff/{position}",
  method: "GET",
  request: z.object({}),
  response: GetReviewDiffResponse,
} as const satisfies Endpoint;
export type GetReviewDiff = typeof GetReviewDiff;

File: gitdot-api/src/endpoint/review/get_review_diff.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::repository::RepositoryDiffFileResource};

pub struct GetReviewDiff;

impl Endpoint for GetReviewDiff {
    const PATH: &'static str = "/repository/{owner}/{repo}/review/{number}/diff/{position}";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetReviewDiffRequest;
    type Response = GetReviewDiffResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct GetReviewDiffRequest {
    pub revision: Option<i32>,
    pub compare_to: Option<i32>,
}

#[derive(ApiResource, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
pub struct GetReviewDiffResponse {
    pub files: Vec<RepositoryDiffFileResource>,
}
